formatter.go

```diff
@@ -170,6 +169,14 @@ func (p *printer) printValue(v reflect.Value, showType, quote bool) {
 				pp = p.indent()
 			}
 			for i := 0; i < v.NumField(); i++ {
+				fieldTag := t.Field(i)
+				val := fieldTag.Tag.Get("pretty")
+				if val == "-" {
+					continue
+				}
+				if !expand && i > 0 {
+					io.WriteString(pp, ", ")
```

Comment:
I don't understand why this existing code moved up here. Would it be possible to leave it where it was?

Reply:
Yeah this was the toughest part to get right. It's not possible to easily leave it where it was, because if *any* fields are ignored, an extra comma *will* be at the end. 

With it moved up here, we are checking if we are not in the first field, because the first field is the only field with no comma at beginning of it.

Another way to implement would be to count ignored fields, but this way not much is changed.

Reply:
Ah I see. Unfortunately even this code has a similar problem. Here's an additional test:

```go
type testFieldTag1 struct {
	z    string `pretty:"-"`
	x, y int
}

func TestStructFieldTags1(t *testing.T) {
	t1 := testFieldTag1{x: 1, y: 2, z: "ignored"}
	got := Sprint(t1)
	expected := "pretty.testFieldTag1{x:1, y:2}"
	if got != expected {
		t.Logf("Got:    %s", got)
		t.Logf("Wanted: %s", expected)
		t.Fail()
	}
}
```

If I run this test against this PR, I get

```
formatter_test.go|316| Got:    pretty.testFieldTag1{, x:1, y:2}                 
formatter_test.go|317| Wanted: pretty.testFieldTag1{x:1, y:2}
```